### PR TITLE
Added message while loading metrics for custom dashboards

### PR DESCRIPTION
**Pull request by hhovsepy**
Issue https://github.com/kiali/kiali/issues/5407

Added message "Loading metrics" while custom dashboards ate not loaded, waiting for that message to disappear in cypress tests.

### CODE REVIEW

File: frontend/src/components/Metrics/CustomMetrics.tsx [hunk 1]
```diff
@@ -226,7 +258,7 @@ class CustomMetrics extends React.Component<Props, MetricsState> {
     const content = (
       <>
         {this.renderOptionsBar()}
-        {this.state.dashboard && dashboard}
+        {this.state.dashboard !== undefined ? dashboard : this.renderFetchMetrics('Loading metrics', 'Metrics will be loaded in a few seconds...')}
```
Comment (nrfox) [inline frontend/src/components/Metrics/CustomMetrics.tsx:261]: Kiali seems to be somewhat inconsistent on how it handles loading states. The graph page shows "Loading Graph" text but the other metrics charts and the list pages (istio config, workload, etc) either show nothing or show text saying something like "No Traces" until the data loads. Personally I prefer the loading text as it at least gives you the expectation that something is happening and will load eventually.

Can you have an empty custom tab with no dashboard? Would you then see "Loading metrics" permanently? If you can have an empty custom tab with no dashboard then there would be three states: dashboard is undefined (no dashboard), loading metrics (dashboard is loading could still return empty and go back to no dashboard), dashboard is defined.

File: frontend/src/components/Metrics/CustomMetrics.tsx [hunk 2]
```diff
@@ -198,6 +217,19 @@ class CustomMetrics extends React.Component<Props, MetricsState> {
...
   }
 
+  renderFetchMetrics = (title, msg) => {
+    return (
+      <div className={emptyStyle}>
+        <EmptyState variant={EmptyStateVariant.small}>
+          <Title headingLevel="h5" size={TitleSizes.lg}>
+            {title}
+          </Title>
+          <EmptyStateBody>{msg}</EmptyStateBody>
```
Comment (nrfox) [inline frontend/src/components/Metrics/CustomMetrics.tsx:227]: imo you don't need the message. Just "Loading metrics" is good enough and there's no guarantee that the metrics will load in a few seconds :). The graph page just has "Loading Graph" so it'd make it more consistent.

**Comment by nrfox** [comment]
Thanks for tackling one of the flakes!

File: frontend/src/components/Metrics/CustomMetrics.tsx (hunk 2 above)
Comment (jmazzitelli) [reply frontend/src/components/Metrics/CustomMetrics.tsx:227]: > there's no guarantee that the metrics will load in a few seconds :)

I agree. Let's not give the user any kind of guarantee or even expectation of load times. What is a few seconds? 2 seconds? 5? 10? 30? 120? :) I would say just tell them its Loading and that's it.

Comment (hhovsepy) [reply frontend/src/components/Metrics/CustomMetrics.tsx:227]: I first had the title only, but visually it seemed strange to have only Header without body. But agree that "few seconds" is odd.

File: frontend/src/components/Metrics/CustomMetrics.tsx (hunk 1 above)
Comment (hhovsepy) [reply frontend/src/components/Metrics/CustomMetrics.tsx]: Imho the message "No metrics" could be confusing, as it shows metrics right after.
When there was a problem it loading or showing custom dashboards, there is a error message in notifications:
Screenshot from 2022-09-09 09-14-07
So it s clear, that metrics were loading but error happened.

**Comment by hhovsepy** [comment]
Thanks for the review. have removed the "few seconds..." message.

File: frontend/src/components/Metrics/CustomMetrics.tsx (hunk 1 above)
Comment (nrfox) [reply frontend/src/components/Metrics/CustomMetrics.tsx]: There's potentially 4 states then!

1. Loading - request for dashboard
2. Dashboard(s) exist - finished loading successfully
3. No dashboard - finished loading successfully but empty dashboard
4. Error in request - finished loading but had an error in the request

It looks like you actually can have the custom tab with no dashboard but this is probably an invalid state and even in the main branch the page is currently blank so state 3 we don't have to worry about and state 4 is handled by the message center and more clear now by having the "Loading message".
Screenshot from 2022-09-09 11-01-22

**Review by nrfox — APPROVED** [review]

**Pull request merged**